### ISSUE TITLE
Cannot set Vector sink to `datadog_logs`

### ISSUE THREAD

**Issue by BeeeQueue** [dokku/dokku#4346]
## Description of problem

I'm trying out the new Vector logs, but when trying to set it up with Datadog it seems to fail to parse the sink DSN.

### How reproducible

100%

### Steps to Reproduce

1. Set up Vector logging
1. Try to set sink to the [Datadog one](https://vector.dev/docs/reference/sinks/datadog_logs/)
   `$ dokku logs:set app vector-sink "datadog_logs://?api_key=abc123"`

#### Actual Results

Error:
` !     parse "datadog_logs://?api_key=abc123": first path segment in URL cannot contain colon`

#### Expected Results

Configuring the sink as expected

### `dokku report APP_NAME` output

```
-----> uname: Linux ds5518 5.4.0-60-generic #67-Ubuntu SMP Tue Jan 5 18:31:36 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
                     total        used        free      shared  buff/cache   available
       Mem:          64291        2064       26809          16       35416       61665
       Swap:          8191           0        8191
-----> docker version:
       Client:
        Version:           19.03.8
        API version:       1.40
        Go version:        go1.13.8
        Git commit:        afacb8b7f0
        Built:             Fri Dec 18 12:15:19 2020
        OS/Arch:           linux/amd64
        Experimental:      false

       Server:
        Engine:
         Version:          19.03.8
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.13.8
         Git commit:       afacb8b7f0
         Built:            Fri Dec  4 23:02:49 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.3.3-0ubuntu2.2
         GitCommit:
        runc:
         Version:          spec: 1.0.1-dev
         GitCommit:
        docker-init:
         Version:          0.18.0
         GitCommit:
-----> docker daemon info:
       Client:
        Debug Mode: true

       Server:
        Containers: 8
         Running: 8
         Paused: 0
         Stopped: 0
        Images: 63
        Server Version: 19.03.8
        Storage Driver: overlay2
         Backing Filesystem: <unknown>
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
WARNING: No swap limit support
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: nvidia runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version:
        runc version:
        init version:
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 5.4.0-60-generic
        Operating System: Ubuntu 20.04.1 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 20
        Total Memory: 62.78GiB
        Name: ds5518
        ID: KFY6:P6D5:GUDY:EGBX:IJ35:PR5T:7NJB:N6A2:5CAB:GET6:J2YW:QRWJ
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false

-----> git version: git version 2.25.1
-----> sigil version: 0.6.0
-----> herokuish version:
       herokuish: 0.5.24
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v222
         heroku-buildpack-nodejs    v182
         heroku-buildpack-clojure   v86
         heroku-buildpack-python    v188
         heroku-buildpack-java      v69
         heroku-buildpack-gradle    v34
         heroku-buildpack-scala     v88
         heroku-buildpack-play      v26
         heroku-buildpack-php       v188
         heroku-buildpack-go        v149
         buildpack-nginx            v14
-----> dokku version: dokku version 0.22.9
-----> dokku plugins:
       plugn: 0.6.1
         00_dokku-standard    0.22.9 enabled    dokku core standard plugin
         20_events            0.22.9 enabled    dokku core events logging plugin
         app-json             0.22.9 enabled    dokku core app-json plugin
         apps                 0.22.9 enabled    dokku core apps plugin
         builder-cnb          0.22.9 enabled    dokku core builder-cnb plugin
         builder-dockerfile   0.22.9 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.22.9 enabled    dokku core builder-herokuish plugin
         buildpacks           0.22.9 enabled    dokku core buildpacks plugin
         certs                0.22.9 enabled    dokku core certificate management plugin
         checks               0.22.9 enabled    dokku core checks plugin
         common               0.22.9 enabled    dokku core common plugin
         config               0.22.9 enabled    dokku core config plugin
         docker-options       0.22.9 enabled    dokku core docker-options plugin
         domains              0.22.9 enabled    dokku core domains plugin
         enter                0.22.9 enabled    dokku core enter plugin
         git                  0.22.9 enabled    dokku core git plugin
         letsencrypt          0.9.4 enabled    Automated installation of let's encrypt TLS certificates
         logs                 0.22.9 enabled    dokku core logs plugin
         monorepo             0.1.0 enabled    Deploy all apps from a monorepo.
         network              0.22.9 enabled    dokku core network plugin
         nginx-vhosts         0.22.9 enabled    dokku core nginx-vhosts plugin
         plugin               0.22.9 enabled    dokku core plugin plugin
         postgres             1.11.5 enabled    dokku postgres service plugin
         proxy                0.22.9 enabled    dokku core proxy plugin
         ps                   0.22.9 enabled    dokku core ps plugin
         redis                1.12.5 enabled    dokku redis service plugin
         repo                 0.22.9 enabled    dokku core repo plugin
         resource             0.22.9 enabled    dokku core resource plugin
         scheduler-docker-local 0.22.9 enabled    dokku core scheduler-docker-local plugin
         shell                0.22.9 enabled    dokku core shell plugin
         ssh-keys             0.22.9 enabled    dokku core ssh-keys plugin
         storage              0.22.9 enabled    dokku core storage plugin
         tags                 0.22.9 enabled    dokku core tags plugin
         tar                  0.22.9 enabled    dokku core tar plugin
         trace                0.22.9 enabled    dokku core trace plugin
=====> anisubs-manager app information
       App deploy source:
       App dir:                       /home/dokku/anisubs-manager
       App locked:                    true
=====> anisubs-manager buildpacks information
       Buildpacks list:
xargs: unmatched single quote; by default quotes are special to xargs unless you use the -0 option
CN = api.anisubs.app
error 20 at 0 depth lookup: unable to get local issuer certificate
=====> anisubs-manager ssl information
       Ssl dir:                       /home/dokku/anisubs-manager/tls
       Ssl enabled:                   true
       Ssl hostnames:                 api.anisubs.app
       Ssl expires at:                Apr 17 13:46:43 2021 GMT
       Ssl issuer:                    C = US, O =
       Ssl starts at:                 Jan 17 13:46:43 2021 GMT
       Ssl subject:                   subject=CN = api.anisubs.app
       Ssl verified:                  self signed
=====> anisubs-manager checks information
       Checks disabled list:          none
       Checks skipped list:           none
=====> anisubs-manager docker options information
       Docker options build:          --link dokku.postgres.manager:dokku-postgres-manager --link dokku.redis.manager:dokku-redis-manager
       Docker options deploy:         --link dokku.postgres.manager:dokku-postgres-manager --link dokku.redis.manager:dokku-redis-manager --restart=on-failure:10 -v /home/docker/anisubs/output:/app/output
       Docker options run:            --link dokku.postgres.manager:dokku-postgres-manager --link dokku.redis.manager:dokku-redis-manager -v /home/docker/anisubs/output:/app/output
=====> anisubs-manager domains information
       Domains app enabled:           true
       Domains app vhosts:            api.anisubs.app
       Domains global enabled:        false
       Domains global vhosts:
=====> anisubs-manager git information
       Git deploy branch:             master
       Git global deploy branch:      master
       Git keep git dir:              false
       Git rev env var:               GIT_REV
       Git sha:                       f893a00
       Git last updated at:
=====> anisubs-manager logs information
       Logs global vector sink:
       Logs vector sink:
=====> anisubs-manager network information
       Network attach post create:
       Network attach post deploy:
       Network bind all interfaces:   false
       Network web listeners:         172.17.0.7:5000
=====> anisubs-manager nginx information
       Nginx access log format:
       Nginx access log path:         /var/log/nginx/anisubs-manager-access.log
       Nginx bind address ipv4:
       Nginx bind address ipv6:       ::
       Nginx disable custom config:   false
       Nginx error log path:          /var/log/nginx/anisubs-manager-error.log
       Nginx hsts:                    true
       Nginx hsts include subdomains: true
       Nginx hsts max age:            15724800
       Nginx hsts preload:            false
       Nginx proxy buffer size:       4096
       Nginx proxy buffering:         on
       Nginx proxy buffers:           8 4096
       Nginx proxy busy buffers size: 8192
       Nginx proxy read timeout:      60s
       Nginx last visited at:         1611317749
=====> anisubs-manager proxy information
       Proxy enabled:                 true
       Proxy port map:                http:80:5000 https:443:5000
       Proxy type:                    nginx
=====> anisubs-manager ps information
       Deployed:                      true
       Processes:                     1
       Ps can scale:                  true
       Ps restart policy:             on-failure:10
       Restore:                       true
       Running:                       true
       Status web 1:                  running (CID: b339a2af5fa)
=====> anisubs-manager scheduler-docker-local information
       Scheduler docker local disable chown:
=====> anisubs-manager storage information
       Storage build mounts:
       Storage deploy mounts:         -v /home/docker/anisubs/output:/app/output
       Storage run mounts:            -v /home/docker/anisubs/output:/app/output
```

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?:

Install script / Physical server

### Additional information

- Output of failing Dokku commands after running `dokku trace:on`:

```
+ export DOKKU_HOST_ROOT=/home/dokku
+ DOKKU_HOST_ROOT=/home/dokku
+ export DOKKU_DISTRO
++ . /etc/os-release
++ echo ubuntu
+ DOKKU_DISTRO=ubuntu
+ export DOCKER_BIN=docker
+ DOCKER_BIN=docker
+ export DOKKU_IMAGE=gliderlabs/herokuish:latest
+ DOKKU_IMAGE=gliderlabs/herokuish:latest
+ export DOKKU_LIB_ROOT=/var/lib/dokku
+ DOKKU_LIB_ROOT=/var/lib/dokku
+ export PLUGIN_PATH=/var/lib/dokku/plugins
+ PLUGIN_PATH=/var/lib/dokku/plugins
+ export PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ export PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ export PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ export PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ export PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ export DOKKU_SYSTEM_GROUP=dokku
+ DOKKU_SYSTEM_GROUP=dokku
+ export DOKKU_SYSTEM_USER=dokku
+ DOKKU_SYSTEM_USER=dokku
+ export DOKKU_API_VERSION=1
+ DOKKU_API_VERSION=1
+ export DOKKU_NOT_IMPLEMENTED_EXIT=10
+ DOKKU_NOT_IMPLEMENTED_EXIT=10
+ export DOKKU_VALID_EXIT=0
+ DOKKU_VALID_EXIT=0
+ export DOKKU_PID=1628607
+ DOKKU_PID=1628607
+ export DOKKU_LOGS_DIR=/var/log/dokku
+ DOKKU_LOGS_DIR=/var/log/dokku
+ export DOKKU_LOGS_HOST_DIR=/var/log/dokku
+ DOKKU_LOGS_HOST_DIR=/var/log/dokku
+ export DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ export DOKKU_CONTAINER_LABEL=dokku
+ DOKKU_CONTAINER_LABEL=dokku
+ export 'DOKKU_GLOBAL_BUILD_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_BUILD_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ export 'DOKKU_GLOBAL_RUN_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_RUN_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ parse_args logs:set anisubs-manager vector-sink 'datadog_logs://?api_key=XXXXXX'
+ declare 'desc=top-level cli arg parser'
+ local next_index=1
+ local skip=false
+ args=("$@")
+ local args
+ local flags
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ logs:set == \-\-\a\p\p ]]
+ [[ logs:set =~ ^--.* ]]
+ next_index=2
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ anisubs-manager == \-\-\a\p\p ]]
+ [[ anisubs-manager =~ ^--.* ]]
+ next_index=3
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ vector-sink == \-\-\a\p\p ]]
+ [[ vector-sink =~ ^--.* ]]
+ next_index=4
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ datadog_logs://?api_key=XXXXXX== \-\-\a\p\p ]]
+ [[ datadog_logs://?api_key=XXXXXX=~ ^--.* ]]
+ next_index=5
+ [[ -z '' ]]
++ echo -e ''
++ sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
+ export DOKKU_GLOBAL_FLAGS=
+ DOKKU_GLOBAL_FLAGS=
+ return 0
+ args=("$@")
+ skip_arg=false
+ [[ logs:set =~ ^--.* ]]
+ has_tty
+ declare 'desc=return 0 if we have a tty'
++ LC_ALL=C
++ /usr/bin/tty
+ [[ /dev/pts/1 == \n\o\t\ \a\ \t\t\y ]]
+ return 0
++ id -un
+ [[ adam != \d\o\k\k\u ]]
+ [[ ! logs:set =~ plugin:* ]]
+ [[ logs:set != \s\s\h\-\k\e\y\s\:\a\d\d ]]
+ [[ logs:set != \s\s\h\-\k\e\y\s\:\r\e\m\o\v\e ]]
++ id -un
+ export SSH_USER=XXXXXX
+ SSH_USER=XXXXXX
+ sudo -u dokku -E -H /usr/bin/dokku logs:set anisubs-manager vector-sink 'datadog_logs://?api_key=XXXXXX'
+ export DOKKU_HOST_ROOT=/home/dokku
+ DOKKU_HOST_ROOT=/home/dokku
+ export DOKKU_DISTRO
++ . /etc/os-release
++ echo ubuntu
+ DOKKU_DISTRO=ubuntu
+ export DOCKER_BIN=docker
+ DOCKER_BIN=docker
+ export DOKKU_IMAGE=gliderlabs/herokuish:latest
+ DOKKU_IMAGE=gliderlabs/herokuish:latest
+ export DOKKU_LIB_ROOT=/var/lib/dokku
+ DOKKU_LIB_ROOT=/var/lib/dokku
+ export PLUGIN_PATH=/var/lib/dokku/plugins
+ PLUGIN_PATH=/var/lib/dokku/plugins
+ export PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ export PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ export PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ export PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ export PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ export DOKKU_SYSTEM_GROUP=dokku
+ DOKKU_SYSTEM_GROUP=dokku
+ export DOKKU_SYSTEM_USER=dokku
+ DOKKU_SYSTEM_USER=dokku
+ export DOKKU_API_VERSION=1
+ DOKKU_API_VERSION=1
+ export DOKKU_NOT_IMPLEMENTED_EXIT=10
+ DOKKU_NOT_IMPLEMENTED_EXIT=10
+ export DOKKU_VALID_EXIT=0
+ DOKKU_VALID_EXIT=0
+ export DOKKU_PID=1628617
+ DOKKU_PID=1628617
+ export DOKKU_LOGS_DIR=/var/log/dokku
+ DOKKU_LOGS_DIR=/var/log/dokku
+ export DOKKU_LOGS_HOST_DIR=/var/log/dokku
+ DOKKU_LOGS_HOST_DIR=/var/log/dokku
+ export DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ export DOKKU_CONTAINER_LABEL=dokku
+ DOKKU_CONTAINER_LABEL=dokku
+ export 'DOKKU_GLOBAL_BUILD_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_BUILD_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ export 'DOKKU_GLOBAL_RUN_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_RUN_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ parse_args logs:set anisubs-manager vector-sink 'datadog_logs://?api_key=XXXXXX'
+ declare 'desc=top-level cli arg parser'
+ local next_index=1
+ local skip=false
+ args=("$@")
+ local args
+ local flags
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ logs:set == \-\-\a\p\p ]]
+ [[ logs:set =~ ^--.* ]]
+ next_index=2
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ anisubs-manager == \-\-\a\p\p ]]
+ [[ anisubs-manager =~ ^--.* ]]
+ next_index=3
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ vector-sink == \-\-\a\p\p ]]
+ [[ vector-sink =~ ^--.* ]]
+ next_index=4
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ datadog_logs://?api_key=XXXXXX== \-\-\a\p\p ]]
+ [[ datadog_logs://?api_key=XXXXXX=~ ^--.* ]]
+ next_index=5
+ [[ -z '' ]]
++ echo -e ''
++ sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
+ export DOKKU_GLOBAL_FLAGS=
+ DOKKU_GLOBAL_FLAGS=
+ return 0
+ args=("$@")
+ skip_arg=false
+ [[ logs:set =~ ^--.* ]]
+ has_tty
+ declare 'desc=return 0 if we have a tty'
++ LC_ALL=C
++ /usr/bin/tty
+ [[ /dev/pts/1 == \n\o\t\ \a\ \t\t\y ]]
+ return 0
++ id -un
+ [[ dokku != \d\o\k\k\u ]]
+ [[ logs:set =~ ^plugin:.* ]]
+ [[ logs:set == \s\s\h\-\k\e\y\s\:\a\d\d ]]
+ [[ logs:set == \s\s\h\-\k\e\y\s\:\r\e\m\o\v\e ]]
+ [[ -n '' ]]
+ dokku_auth logs:set anisubs-manager vector-sink 'datadog_logs://?api_key=XXXXXX'
+ declare 'desc=calls user-auth plugin trigger'
+ export SSH_USER=XXXXXX
+ SSH_USER=XXXXXX
+ export SSH_NAME=default
+ SSH_NAME=default
++ wc -l
++ find /var/lib/dokku/plugins/enabled/20_events/user-auth
+ [[ 1 == 1 ]]
+ return 0
+ case "$1" in
+ execute_dokku_cmd logs:set anisubs-manager vector-sink 'datadog_logs://?api_key=XXXXXX'
+ declare 'desc=executes dokku sub-commands'
+ local PLUGIN_NAME=logs:set
+ local PLUGIN_CMD=logs:set
+ local implemented=0
+ local script
+ argv=("$@")
+ local argv
+ case "$PLUGIN_NAME" in
++ readlink -f /var/lib/dokku/plugins/enabled/logs
+ [[ /var/lib/dokku/core-plugins/available/logs == *core-plugins* ]]
+ [[ logs:set == \l\o\g\s\:\s\e\t ]]
+ shift 1
+ [[ -n '' ]]
+ set -- logs:set anisubs-manager vector-sink 'datadog_logs://?api_key=XXXXXX'
+ [[ -x /var/lib/dokku/plugins/enabled/logs:set/subcommands/default ]]
+ [[ -x /var/lib/dokku/plugins/enabled/logs:set/subcommands/logs:set ]]
+ [[ -x /var/lib/dokku/plugins/enabled/logs/subcommands/set ]]
+ [[ -n set ]]
+ /var/lib/dokku/plugins/enabled/logs/subcommands/set logs:set anisubs-manager vector-sink 'datadog_logs://?api_key=XXXXXX'
 !     parse "datadog_logs://?api_key=XXXXXX": first path segment in URL cannot contain colon
```

**Comment by josegonzalez** [comment]
Seems like a fun golang bug. I think we can hack this by replacing it with `-` maybe and then re-replacing afterwards? Will have to check...

Thanks for the report!

**Comment by josegonzalez** [comment]
Closed by #4355.
